src/arch/arm/isa/insts/fp.isa

```diff
@@ -485,14 +487,19 @@ let {{
         finishVfp(fpscr, state, fpscr.fz);
         FpscrExc = fpscr;
     '''
-    singleBinOp = "binaryOp(fpscr, FpOp1, FpOp2," + \
-                "%(func)s, fpscr.fz, fpscr.dn, fpscr.rMode)"
-    singleUnaryOp = "unaryOp(fpscr, FpOp1, %(func)s, fpscr.fz, fpscr.rMode)"
+    singleBinOp = "floatToBits32(binaryOp(fpscr, FpOp1, FpOp2, " + \
+                  "%(func)s, fpscr.fz, fpscr.dn, fpscr.rMode))"
+    singleUnaryOp = "floatToBits32(unaryOp(fpscr, FpOp1, " + \
+                    "%(func)s, fpscr.fz, fpscr.rMode))"
     doubleCode = vfpEnabledCheckCode + '''
         [[maybe_unused]] FPSCR fpscr = (FPSCR) FpscrExc;
-        double dest = %(op)s;
-        FpDestP0_uw = dblLow(dest);
-        FpDestP1_uw = dblHi(dest);
+        [[maybe_unused]] uint64_t cOp1 = ((uint64_t)FpOp1P0_uw |
+                                          ((uint64_t)FpOp1P1_uw << 32));
```

Comment:
I understand adding this here avoids some repetition, but we need to be very careful about introducing unnecessary reads. This is not only a simulator performance problem; when we use these magic operands, we inform the CPU the instruction has a Read dependency over that register. This will affect the dependency graph and it will pollute dynamic scheduling with some unnecessary serialization (If a RAW is present).

Only code making use of FpOp1P0 and FpOp1P1 should read them and the maybe_unused scares me a bit.

Reply:
I will update this PR to avoid maybe_unused by adding this call to the vrint() helpers.